doc/coding-ghost.md

```diff
@@ -0,0 +1,22 @@
+# Getting started with gh-ost development.
+
+## Overview
+
+Getting started with gh-ost development is simple!
+
+- First clone the repository.
+- From inside of the repository run `script/cibuild`
+- This will bootstrap the environment if needed, format the code, build the code, and then run the unit test.
+
+## CI build workflow
+
+`script/cibuild` performs the following actions:
+
+- It runs `script/bootstrap`
+- `script/bootstrap` runs `script/ensure-go-installed`
+- `script/ensure-go-installed` installs go locally if (go is not installed) || (go is not version 1.7). It will not install go if it is already installed locally and at the correct version.
+- `script/build` builds the `gh-ost` binary and places in in `bin/`
```

Comment:
I think enumerating the exact steps taken by the script is error prone. e.g. what happens if the script ensures go `1.8` -- will this text be updated?